unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/ParcelScene.cs

```diff
@@ -1174,94 +1114,51 @@ private DecentralandEntity GetEntityForUpdate(string entityId)
+            foreach (string id in allDisposableComponents)
             {
-                SetSceneReady();
+                Environment.i.parcelScenesCleaner.MarkDisposableComponentForCleanup(this, id);
             }
-
-            OnStateRefreshed?.Invoke(this);
-            RefreshName();
         }
 
-        public void SetInitMessagesDone()
+        public string GetStateString()
         {
-            if (isReleased)
-                return;
-
-            if (state == State.READY)
+            switch (sceneLifecycleHandler.state)
             {
-                Debug.LogWarning($"Init messages done after ready?! {sceneData.basePosition}", gameObject);
-                return;
+                case SceneLifecycleHandler.State.NOT_READY:
+                    return $"scene:{prettyName} - not ready...";
+                case SceneLifecycleHandler.State.WAITING_FOR_INIT_MESSAGES:
+                    return $"scene:{prettyName} - waiting for init messages...";
+                case SceneLifecycleHandler.State.WAITING_FOR_COMPONENTS:
+                    if (disposableComponents != null && disposableComponents.Count > 0)
+                        return $"scene:{prettyName} - left to ready:{disposableComponents.Count - sceneLifecycleHandler.disposableNotReadyCount}/{disposableComponents.Count}";
+                    else
+                        return $"scene:{prettyName} - no components. waiting...";
+                case SceneLifecycleHandler.State.READY:
+                    return $"scene:{prettyName} - ready!";
             }
 
-            state = State.WAITING_FOR_COMPONENTS;
-            RefreshName();
-
-            if (disposableNotReadyCount > 0)
-            {
-                //NOTE(Brian): Here, we have to split the iterations. If not, we will get repeated calls of
-                //             SetSceneReady(), as the disposableNotReady count is 1 and gets to 0
-                //             in each OnDisposableReady() call.
-
-                using (var iterator = disposableComponents.GetEnumerator())
-                {
-                    while (iterator.MoveNext())
-                    {
-                        disposableComponents[iterator.Current.Value.id].CallWhenReady(OnDisposableReady);
-                    }
-                }
-            }
-            else
-            {
-                SetSceneReady();
-            }
+            return $"scene:{prettyName} - no state?";
         }
 
-        private void SetSceneReady()
+        public void RefreshName()
         {
-            if (state == State.READY)
-                return;
-
-            if (VERBOSE)
-                Debug.Log($"{sceneData.basePosition} Scene Ready!");
-
-            state = State.READY;
-
-            Environment.i.sceneController.SendSceneReady(sceneData.id);
-            RefreshName();
-
-            OnSceneReady?.Invoke(this);
+#if UNITY_EDITOR
+            Debug.Log(GetStateString());
+#endif
```

Comment:
We could still apply the state string to the GO name right? in that case we should add that here, maybe even replace this `Debug.Log()` with that